src/main/kotlin/org/partiql/scribe/targets/spark/SparkExpandStruct.kt

```diff
@@ -0,0 +1,191 @@
+package org.partiql.scribe.targets.spark
+
+import org.partiql.plan.Fn
+import org.partiql.plan.Rex
+import org.partiql.plan.rex
+import org.partiql.plan.rexOpCallStatic
+import org.partiql.plan.rexOpLit
+import org.partiql.plan.rexOpPathKey
+import org.partiql.plan.rexOpStruct
+import org.partiql.plan.rexOpStructField
+import org.partiql.scribe.asNonNullable
+import org.partiql.types.CollectionType
+import org.partiql.types.StaticType
+import org.partiql.types.StructType
+import org.partiql.types.function.FunctionParameter
+import org.partiql.types.function.FunctionSignature
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.PartiQLValueType
+import org.partiql.value.stringValue
+import org.partiql.value.symbolValue
+
+private fun StaticType.expand() = this.metas["EXPAND"] == true
+
+internal fun StaticType.toRexSpark(prefixPath: Rex): Rex {
+    if (!this.expand()) {
+        return prefixPath
+    }
+    return when (val nonNullType = this.asNonNullable()) {
+        is StructType -> Rex(
+            type = nonNullType,
+            op = when (nonNullType.fields.size) {
+                0 -> error("Currently do not support outputting empty structs in Spark. Consider `EXCLUDE` on the outer struct.")
```

Comment:
(self-review) For Spark struct reconstruction, now give an error due to the confusing behavior of empty structs in Spark -- https://github.com/partiql/partiql-scribe/issues/65.